### ISSUE TITLE
python 3.8 "RecursionError: maximum recursion depth exceeded" for CIMInt.__repr__ 

### ISSUE THREAD

**Issue by tasleson** [pywbem/pywbem#1913]
I'm working on automated unit tests against installed pywbem package during package creation.  The stack trace below only happens on fedora rawhide.  If I run on F30 it runs without error.  The included bash script is what I'm executing to exercise the unit tests.  It assumes we have the pywbem upstream git source repo cloned which is passed as its only argument.  It runs the unit tests against the newly built and installed pywbem package.

I'm going to investigate a bit more, but I thought I would document here in case anyone else as encountered it already.  It appears to be 100% reproducible.

```
___________________________________________________ ERROR collecting tests/unittest/pywbem/test_cim_obj.py ___________________________________________________
.virtualenv/gate/lib/python3.8/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
.virtualenv/gate/lib/python3.8/site-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.virtualenv/gate/lib/python3.8/site-packages/pluggy/manager.py:83: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:234: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:403: in _genfunctions
    self.ihook.pytest_generate_tests.call_extra(methods, dict(metafunc=metafunc))
.virtualenv/gate/lib/python3.8/site-packages/pluggy/hooks.py:324: in call_extra
    return self(**kwargs)
.virtualenv/gate/lib/python3.8/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
.virtualenv/gate/lib/python3.8/site-packages/pluggy/manager.py:92: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.virtualenv/gate/lib/python3.8/site-packages/pluggy/manager.py:83: in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:129: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:975: in parametrize
    ids = self._resolve_arg_ids(argnames, ids, parameters, item=self.definition)
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:1029: in _resolve_arg_ids
    ids = idmaker(argnames, parameters, idfn, ids, self.config, item=item)
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:1183: in idmaker
    ids = [
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:1184: in <listcomp>
    _idvalset(valindex, parameterset, argnames, idfn, ids, config=config, item=item)
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:1173: in _idvalset
    this_id = [
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:1174: in <listcomp>
    _idval(val, argname, idx, idfn, item=item, config=config)
.virtualenv/gate/lib/python3.8/site-packages/_pytest/python.py:1159: in _idval
    return str(val)
/usr/lib/python3.8/site-packages/pywbem/cim_types.py:823: in __repr__
    return _format(
/usr/lib/python3.8/site-packages/pywbem/_utils.py:287: in _format
    return _ASCII2_FORMATTER.format(format_str, *args, **kwargs)
/usr/lib64/python3.8/string.py:163: in format
    return self.vformat(format_string, args, kwargs)
/usr/lib64/python3.8/string.py:167: in vformat
    result, _ = self._vformat(format_string, args, kwargs, used_args, 2)
/usr/lib64/python3.8/string.py:220: in _vformat
    result.append(self.format_field(obj, format_spec))
/usr/lib64/python3.8/string.py:237: in format_field
    return format(value, format_spec)
/usr/lib/python3.8/site-packages/pywbem/cim_types.py:823: in __repr__
    return _format(
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
```

Script setting up env and running tests.
```bash
#!/bin/bash

# Run the existing unit tests against the installed package.

function _chk
{
    echo "executing: $*"
    eval "$@"
    local rc=$?
    if [ $rc -ne 0 ]; then
        mv pywbem_disabled pywbem
        echo "exiting on error ($rc)!"
        exit 1
    fi
}

if [ -z "$1" ]; then
    echo "syntax: unit_test.sh <pywbem> directory"
    exit 1
fi

_chk cd "$1"

# Switch upstream repo to match installed packaged version
VER=$(rpm -q --queryformat '%{VERSION}' python3-pywbem)
_chk git checkout "$VER"

_chk virtualenv --system-site-packages --no-wheel .virtualenv/gate

_chk source .virtualenv/gate/bin/activate

export TEST_INSTALLED=1
export TEST_WBEMCLI_NAME=pywbemcli
export TESTOPTS="-v -x"

_chk make clobber
_chk make develop

# Ensure we have pywbem installed in system
_chk mv pywbem pywbem_disabled
_chk make test
_chk mv pywbem_disabled pywbem

exit 0


```

**Comment by tasleson** [comment]
Easy to reproduce with:

``` python
#!/usr/bin/python3

import pywbem
a = pywbem.Uint8(10)
print("%s" % repr(a))
```

```
Traceback (most recent call last):
  File "what.py", line 5, in <module>
    print("%s" % repr(a))
  File "/usr/lib/python3.8/site-packages/pywbem/cim_types.py", line 823, in __repr__
    return _format(
  File "/usr/lib/python3.8/site-packages/pywbem/_utils.py", line 287, in _format
    return _ASCII2_FORMATTER.format(format_str, *args, **kwargs)
  File "/usr/lib64/python3.8/string.py", line 163, in format
    return self.vformat(format_string, args, kwargs)
  File "/usr/lib64/python3.8/string.py", line 167, in vformat
    result, _ = self._vformat(format_string, args, kwargs, used_args, 2)
  File "/usr/lib64/python3.8/string.py", line 220, in _vformat
    result.append(self.format_field(obj, format_spec))
  File "/usr/lib64/python3.8/string.py", line 237, in format_field
    return format(value, format_spec)
  File "/usr/lib/python3.8/site-packages/pywbem/cim_types.py", line 823, in __repr__
    return _format(
  File "/usr/lib/python3.8/site-packages/pywbem/_utils.py", line 287, in _format
    return _ASCII2_FORMATTER.format(format_str, *args, **kwargs)
  File "/usr/lib64/python3.8/string.py", line 163, in format
    return self.vformat(format_string, args, kwargs)
  File "/usr/lib64/python3.8/string.py", line 167, in vformat
    result, _ = self._vformat(format_string, args, kwargs, used_args, 2)
  File "/usr/lib64/python3.8/string.py", line 220, in _vformat
    result.append(self.format_field(obj, format_spec))
  File "/usr/lib64/python3.8/string.py", line 237, in format_field
    return format(value, format_spec)
  File "/usr/lib/python3.8/site-packages/pywbem/cim_types.py", line 823, in __repr__
    return _format(
  File "/usr/lib/python3.8/site-packages/pywbem/_utils.py", line 287, in _format
    return _ASCII2_FORMATTER.format(format_str, *args, **kwargs)
  File "/usr/lib64/python3.8/string.py", line 163, in format
    return self.vformat(format_string, args, kwargs)
  File "/usr/lib64/python3.8/string.py", line 167, in vformat
    result, _ = self._vformat(format_string, args, kwargs, used_args, 2)
  File "/usr/lib64/python3.8/string.py", line 220, in _vformat
    result.append(self.format_field(obj, format_spec))
  File "/usr/lib64/python3.8/string.py", line 237, in format_field
    return format(value, format_spec)
...
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by tasleson** [comment]
Also get the same exception when you try:

```python
a = pywbem.Uint8(10)
print("%s" % str(a))
```

See: https://bugs.python.org/issue36793

**Comment by tasleson** [comment]
This fixes this specific error, however it looks like we have a number of other issues to address from the python 3.8 changes.

```diff
diff --git a/pywbem/cim_types.py b/pywbem/cim_types.py
index 4c50275..1b3e61d 100644
--- a/pywbem/cim_types.py
+++ b/pywbem/cim_types.py
@@ -796,6 +796,8 @@ class CIMInt(CIMType, _Longint):
     #: integer data types.
     maxvalue = None
 
+    _str__ = _Longint.__repr__
+
     def __new__(cls, *args, **kwargs):
 
         # Python 3.7 removed support for passing the value for int() as a
```

**Comment by andy-maier** [comment]
Tony fixed this in PR #1914, which got picked into PR #1893 for master and TBD for 0.15.0
